dev-tools/merge_streamline_pr.py

```diff
@@ -0,0 +1,270 @@
+#!/usr/bin/env python
+
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Utility for creating well-formed pull request merges and pushing them to Streamline.
+#   usage: ./merge_streamline_pr.py    (see config env vars below)
+#
+# This script is inspired by Spark merge script and also borrow some codes from Kafka.
+#
+# This utility assumes you already have a local Streamline git folder and that you
+# have added remotes corresponding to both (i) the pull repository (which pull requests are available)
+# and (ii) the push repository.
+
+import json
+import os
+import subprocess
+import sys
+import urllib2
+
+# Location of your Streamline git development area
+STREAMLINE_HOME = os.environ.get("STREAMLINE_HOME", os.getcwd())
+# Remote name which points to pull repository
+PR_REMOTE_NAME = os.environ.get("PR_REMOTE_NAME", "pull-repo")
```

Comment:
Can the script fail if the PR_REMOTE_NAME is not set and the pull-repo also does not exist ? Right now the script fails while running the git commands, but it will be better to check it and fail at the begining. Similarly for PUSH_REMOTE_NAME.

Reply:
@arunmahadevan Addressed.